.atomist/editors/ConvertManifestToPackageJson.ts

```diff
@@ -21,81 +21,102 @@ import { Pattern } from "@atomist/rug/operations/RugOperation";
 
 import * as yaml from "js-yaml";
 
+export const packageJsonPath = ".atomist/package.json";
+const manifestYamlPath = ".atomist/manifest.yml";
+
+/**
+ * Convert an Atomist manifest.yml into a package.json.
+ */
 @Editor("ConvertManifestToPackageJson",
     "converts an existing Rug archive manifest.yml to a package.json")
 @Tags("rug", "atomist")
 export class ConvertManifestToPackageJson implements EditProject {
 
-    private packageJsonName = ".atomist/package.json";
-    private manifestYamlName = ".atomist/manifest.yml";
-
     public edit(project: Project) {
-        if (project.fileExists(this.manifestYamlName)) {
-            const manifestContent = project.findFile(this.manifestYamlName).content;
-            const manifestYaml = yaml.load(manifestContent);
-            const packageJson = this.loadPackageJson(project);
-
-            const newPackageJson: any = {};
-            // First we add name and version to the top of the package.json
-            newPackageJson.name = `@${manifestYaml.group}/${manifestYaml.artifact}`;
-            newPackageJson.version = manifestYaml.version;
-
-            // Then we copy all existing keys into the new package.json
-            // tslint:disable-next-line:forin
-            for (const key in packageJson) {
-                newPackageJson[key] = packageJson[key];
-            }
+        if (!project.fileExists(manifestYamlPath)) {
+            console.log(`Project ${project.name} does not have a ${manifestYamlPath}`);
+            return;
+        }
 
-            // Create or retrieve atomist section from package.json
-            let atomist: any = {};
-            if (newPackageJson.atomist == null) {
-                newPackageJson.atomist = atomist;
-            } else {
-                atomist = newPackageJson.atomist;
-            }
+        const manifestContent = project.findFile(manifestYamlPath).content;
+        const manifestYaml = yaml.load(manifestContent);
+        const packageJson = loadPackageJson(project);
 
-            atomist.requires = manifestYaml.requires;
+        const newPackageJson = customizePackageJson(project, manifestYaml.artifact,
```

Comment:
I actually coded up something that parsed in the backing archive one, stripped out the things we don't want and then merged them in an earlier version. I can do something like that again.